src/layer/riscv/layernorm_riscv.cpp

```diff
@@ -0,0 +1,328 @@
+// Copyright 2024 Tencent
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "layernorm_riscv.h"
+#include <math.h>
+
+#if __riscv_vector
+#include <riscv_vector.h>
+#include "riscv_usability.h"
+#endif // __riscv_vector
+
+#include "cpu.h"
+
+namespace ncnn {
+
+LayerNorm_riscv::LayerNorm_riscv()
+{
+#if __riscv_vector
+    support_packing = true;
+#endif // __riscv_vector
+#if NCNN_ZFH
+#if __riscv_vector
+    support_fp16_storage = cpu_support_riscv_zvfh();
+#else
+    support_fp16_storage = cpu_support_riscv_zfh();
+#endif
+#endif
+}
+
+#if __riscv_vector
+static inline int layernorm_rvv_pack1_procedure(int size, float* ptr, const float* gamma_data, const float* beta_data, float eps, int affine)
+{
+    float sum = 0.f;
+    float sqsum = 0.f;
+    size_t vl_max = __riscv_vsetvlmax_e32m1();
+    vfloat32m1_t _sum = __riscv_vfmv_s_f_f32m1(0.f, vl_max);
+    vfloat32m1_t _sqsum = __riscv_vfmv_s_f_f32m1(0.f, vl_max);
+
+    {
```

Comment:
reduce procedure is not optimal, do not `vfredsum` in loop body

refer to

https://github.com/Tencent/ncnn/blob/1289458569b68528ebcf6c2881af83413d186f21/src/layer/riscv/softmax_riscv.cpp#L49